src/pandoc-export.ts

```diff
@@ -0,0 +1,109 @@
+'use strict';
+import * as vscode from 'vscode';
+import * as util from 'util';
+import * as child_process from 'child_process';
+import * as fs from 'fs';
+import * as path from 'path';
+
+const execSync = child_process.execSync;
+
+const userTemplateFilename = getUserTemplateFilename();
+let extensionTemplateFilename;
+
+export function pandocExportAsPdf(context: vscode.ExtensionContext) {
+    function getPandocCommand() {
+        // TODO: make setting for this
+        const pandocVariables = [
+            'colorlinks',
+            'linestretch=1.2',
+            'fontsize=12pt',
+            'geometry:left=3cm',
+            'geometry:right=3cm',
+            'geometry:top=2cm',
+            'geometry:bottom=3cm'
+        ];
+        const pandocArgs = ['--toc'];
+
+        const pandocStr = pandocVariables.length > 1 ? ' -V ' + pandocVariables.join(' -V ') : '';
+
+        let templateFilename;
+        if (fs.existsSync(userTemplateFilename)) {
+            templateFilename = userTemplateFilename;
+        } else {
+            extensionTemplateFilename = context.asAbsolutePath('./pandocTemplate.latex');
+            templateFilename = extensionTemplateFilename;
+        }
+        const pandocCommand = `pandoc ${pandocArgs.join(
+            ' '
+        )} --template="${templateFilename}" ${pandocStr} "${documentBaseName}.org" -o "${tempDir + documentName}.tex"`;
+
+        console.log(pandocCommand);
+        return pandocCommand;
+    }
+    function modifyTexFile() {
+        let data = fs.readFileSync(`${tempDir + documentName}.tex`).toString();
+
+        data = data.replace(/(TODO|DONE|WAIT|PRGS)/g, '\\statusBadge{$1}');
+        data = data.replace(/definecolor{color\\statusBadge{(TODO|PRGS|WAIT|DONE)}}/g, 'definecolor{color$1}');
+        data = data.replace(/{\[}(\d\d\d\d-\d?\d-\d?\d\\?\s+\w\w\w){\]}/g, '\\timeStamp{$1}');
+        // data = data.replace(/\\section{/g, '\\clearpage\\section{');
+
+        fs.writeFileSync(`${tempDir + documentName}.tex`, data, 'utf8');
+        return;
+    }
+
+    const documentPath = vscode.window.activeTextEditor.document.uri.fsPath;
+    if (!documentPath.match(/\.org$/)) {
+        vscode.window.showErrorMessage('Cannot pandoc export from non Org-Mode documents');
+        throw TypeError('Wrong document type active');
+    }
+    const documentBaseName = documentPath.replace(/\.org$/, '');
+    const documentDir = documentBaseName.replace(/\/[^\/]+$/, '');
+    const documentName = documentBaseName.replace(/^.*\//, '');
+    const tempDir = documentDir + '/' + `_temp${+new Date() % 1000000}` + '/';
+
+    fs.mkdirSync(tempDir);
+
+    execSync(getPandocCommand());
+
+    modifyTexFile();
+
+    execSync(`latexmk -cd -pdf -interaction=nonstopmode "${tempDir + documentName}.tex"`);
+
+    execSync(`latexmk "${tempDir + documentName}.tex" -cd -c`);
```

Comment:
error should be handled and messages should be printed to the user

Reply:
I added some basic try/catch+window.showErrorMessage) them on your branch in my labtop and it help me to have it working (meaning, installing `pandoc`, `tex` and `latexmk`)